src/cmd_line/commands/tab.ts

```diff
@@ -43,7 +43,7 @@ export class TabCommand extends node.CommandBase {
     }
   }
 
-  execute(): void {
+  async execute(): Promise<void> {
```

Comment:
Agreed that we should make this `async`. However, can you update the other return paths to also return a promise?

Reply:
@jpoon I `await`ed everything I could with the following two exceptions:
1. `await vscode.commands.executeCommand('workbench.action.openEditorAtIndex1');` never resolves. I think it may be a bug with VSCode, so I should reproduce it in a small extension and open an issue on the VSCode repo.
2. `vscode.commands.executeCommand('activeEditorMove')` needs to be updated to `vscode.commands.executeCommand('moveActiveEditor')`. It isn't quite a simple name replacement, so it should probably be done in a different PR.

Reply:
Filed https://github.com/VSCodeVim/Vim/issues/2401 for the refactoring.